{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "#Transrate\n",
    "##Exploring the new quality assessment for de novo transcriptome assemblies. \n",
    "\n",
    "<img src=\"images/Logo.png\">\n",
    "\n",
    "Website here: http://hibberdlab.com/transrate/index.html\n",
    "\n",
    "Transrate trys to address the major issues that arise during de novo assembly. \n",
    "\n",
    "<img src=\"images/Misassembly.png\">\n",
    "\n",
    "###Basic pipeline\n",
    "####Input assembly contigs, paired-end reads\n",
    "<img src=\"images/Pipeline.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Downloading transrate\n",
    "It is an easy enough download. There are two download options:\n",
    "1. Install from source. \n",
    "2. Install with gem (sort of like pip but for ruby). \n",
    "\n",
    "I found that option 1 worked and option 2 did not work. As an aside, you have to run transrate on a 64 bit system, so make sure that you have the correct system setup. \n",
    "\n",
    "``` bash \n",
    "wget https://bintray.com/artifact/download/blahah/generic/transrate-1.0.1-osx.tar.gz\n",
    "tar -xvzf transrate-1.0.1-osx.tar.gz \n",
    "cd transrate-1.0.1-osx\n",
    "\n",
    "```\n",
    "Now all you need to do is edit your PATH to include the transrate folder. This can be done by editing either your .bash_profile or .bashrc file in your home directory. Adding the folowing to the end should suffice:\n",
    "\n",
    "``` bash \n",
    "export PATH=\"/PATH/TO/FOLDER/transrate-1.0.1-osx:$PATH\"\n",
    "\n",
    "```\n",
    "Transrate has a nice feature which will install of the dependencies for you. To do this run: \n",
    "\n",
    "``` bash \n",
    "sudo transrate --install-deps read\n",
    "\n",
    "```\n",
    "It should return the following message. \n",
    "``` bash \n",
    "Checking dependencies\n",
    "All dependences installed\n",
    "```\n",
    "\n",
    "###Running transrate\n",
    "There are three levels of transrate metrics:\n",
    "1. Input *only* the assembly and get basic stats back (akin to TrinityStats.pl... so we will skip it)\n",
    "2. Input the assembly and your reads and get a more detailed quantitative explanation back\n",
    "3. Input the assembly and a \"closely related\" genome \n",
    "\n",
    "###Let's work with the transcriptome example from transrate\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ INFO] 2015-07-13 13:05:36 : Loading assembly: /Users/harrietalexander/Analysis/Transrate/example_data/transcripts.fa\n",
      "[ INFO] 2015-07-13 13:05:36 : Analysing assembly: /Users/harrietalexander/Analysis/Transrate/example_data/transcripts.fa\n",
      "[ INFO] 2015-07-13 13:05:36 : Results will be saved in /Users/harrietalexander/Analysis/Transrate/example_data/transrate_results/transcripts\n",
      "[ INFO] 2015-07-13 13:05:36 : Calculating contig metrics...\n",
      "[ INFO] 2015-07-13 13:05:36 : Contig metrics:\n",
      "[ INFO] 2015-07-13 13:05:36 : -----------------------------------\n",
      "[ INFO] 2015-07-13 13:05:36 : n seqs                           15\n",
      "[ INFO] 2015-07-13 13:05:36 : smallest                        849\n",
      "[ INFO] 2015-07-13 13:05:36 : largest                        2396\n",
      "[ INFO] 2015-07-13 13:05:36 : n bases                       28562\n",
      "[ INFO] 2015-07-13 13:05:36 : mean len                    1904.13\n",
      "[ INFO] 2015-07-13 13:05:36 : n under 200                       0\n",
      "[ INFO] 2015-07-13 13:05:36 : n over 1k                        14\n",
      "[ INFO] 2015-07-13 13:05:36 : n over 10k                        0\n",
      "[ INFO] 2015-07-13 13:05:36 : n with orf                       15\n",
      "[ INFO] 2015-07-13 13:05:36 : mean orf percent              46.46\n",
      "[ INFO] 2015-07-13 13:05:36 : n90                            1612\n",
      "[ INFO] 2015-07-13 13:05:36 : n70                            1681\n",
      "[ INFO] 2015-07-13 13:05:36 : n50                            2037\n",
      "[ INFO] 2015-07-13 13:05:36 : n30                            2288\n",
      "[ INFO] 2015-07-13 13:05:36 : n10                            2385\n",
      "[ INFO] 2015-07-13 13:05:36 : gc                             0.53\n",
      "[ INFO] 2015-07-13 13:05:36 : gc skew                       -0.05\n",
      "[ INFO] 2015-07-13 13:05:36 : at skew                        0.06\n",
      "[ INFO] 2015-07-13 13:05:36 : cpg ratio                      1.59\n",
      "[ INFO] 2015-07-13 13:05:36 : bases n                           0\n",
      "[ INFO] 2015-07-13 13:05:36 : proportion n                    0.0\n",
      "[ INFO] 2015-07-13 13:05:36 : linguistic complexity          0.33\n",
      "[ INFO] 2015-07-13 13:05:36 : Contig metrics done in 0 seconds\n",
      "[ INFO] 2015-07-13 13:05:36 : Calculating read diagnostics...\n",
      "[ INFO] 2015-07-13 13:05:39 : Read mapping metrics:\n",
      "[ INFO] 2015-07-13 13:05:39 : -----------------------------------\n",
      "[ INFO] 2015-07-13 13:05:39 : fragments                     10000\n",
      "[ INFO] 2015-07-13 13:05:39 : fragments mapped              10000\n",
      "[ INFO] 2015-07-13 13:05:39 : p fragments mapped              1.0\n",
      "[ INFO] 2015-07-13 13:05:39 : good mappings                  9993\n",
      "[ INFO] 2015-07-13 13:05:39 : p good mapping                  1.0\n",
      "[ INFO] 2015-07-13 13:05:39 : bad mappings                      7\n",
      "[ INFO] 2015-07-13 13:05:39 : potential bridges                 0\n",
      "[ INFO] 2015-07-13 13:05:39 : bases uncovered                2786\n",
      "[ INFO] 2015-07-13 13:05:39 : p bases uncovered               0.1\n",
      "[ INFO] 2015-07-13 13:05:39 : contigs uncovbase                12\n",
      "[ INFO] 2015-07-13 13:05:39 : p contigs uncovbase             0.8\n",
      "[ INFO] 2015-07-13 13:05:39 : contigs uncovered                 2\n",
      "[ INFO] 2015-07-13 13:05:39 : p contigs uncovered            0.13\n",
      "[ INFO] 2015-07-13 13:05:39 : contigs lowcovered                9\n",
      "[ INFO] 2015-07-13 13:05:39 : p contigs lowcovered            0.6\n",
      "[ INFO] 2015-07-13 13:05:39 : contigs segmented                 0\n",
      "[ INFO] 2015-07-13 13:05:39 : p contigs segmented             0.0\n",
      "[ INFO] 2015-07-13 13:05:39 : Read metrics done in 4 seconds\n",
      "[ INFO] 2015-07-13 13:05:39 : No reference provided, skipping comparative diagnostics\n",
      "[ INFO] 2015-07-13 13:05:39 : TRANSRATE ASSEMBLY SCORE     0.6623\n",
      "[ INFO] 2015-07-13 13:05:39 : -----------------------------------\n",
      "[ INFO] 2015-07-13 13:05:39 : TRANSRATE OPTIMAL SCORE      0.9006\n",
      "[ INFO] 2015-07-13 13:05:39 : TRANSRATE OPTIMAL CUTOFF     0.7652\n",
      "[ INFO] 2015-07-13 13:05:39 : good contigs                     14\n",
      "[ INFO] 2015-07-13 13:05:39 : p good contigs                 0.93\n",
      "[ INFO] 2015-07-13 13:05:39 : Writing contig metrics for each contig to /Users/harrietalexander/Analysis/Transrate/example_data/transrate_results/transcripts/contigs.csv\n",
      "[ INFO] 2015-07-13 13:05:39 : Writing analysis results to assemblies.csv\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "transrate --assembly transcripts.fa --left left.fq --right right.fq "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##What does the output mean? \n",
    "\n",
    "\n",
    "\n",
    "###Basic stats on transcriptome contigs (ignoring mapping)\n",
    "\n",
    "<img src=\"images/Table.png\">\n",
    "\n",
    "##Transrate-specific metrics\n",
    "\n",
    "###**Contig score**: \n",
    "\n",
    "Considers... \n",
    "1. Whether each base has been called correctly \n",
    "2. Whether each base is truly part of the transcript\n",
    "3. The probability that the contig is derived from a single transcript (rather than pieces of two or more transcripts)\n",
    "4. The probability that the contig is structurally complete and correct.\n",
    "\n",
    "###**Transrate score**: \n",
    "\n",
    "Geometric mean of all of the contig scores\n",
    "\n",
    "###Stats on transcriptome contigs (considering mapping)\n",
    "\n",
    "\n",
    "<img src=\"images/Table2.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###What files were produced? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 5112\n",
      "-rw-r--r--  1 harrietalexander  staff    10343 Jul 14 09:03 TransrateLesson.ipynb\n",
      "drwxr-xr-x  9 harrietalexander  staff      306 Jul 14 08:45 images\n",
      "drwxr-xr-x  2 harrietalexander  staff       68 Jul 13 11:38 ipythonNotebook\n",
      "-rw-r--r--@ 1 harrietalexander  staff  1283063 Feb  9  2014 left.fq\n",
      "-rw-r--r--@ 1 harrietalexander  staff  1283063 Feb  9  2014 right.fq\n",
      "-rw-r--r--@ 1 harrietalexander  staff    29213 Feb  9  2014 transcripts.fa\n",
      "drwxr-xr-x  4 harrietalexander  staff      136 Jul 13 13:03 transrate_results\n",
      "-rw-r--r--  1 harrietalexander  staff        0 Jul 13 11:32 untitled.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 8\n",
      "-rw-r--r--   1 harrietalexander  staff  813 Jul 13 13:05 assemblies.csv\n",
      "drwxr-xr-x  14 harrietalexander  staff  476 Jul 13 13:05 transcripts\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l transrate_results/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 2880\n",
      "-rw-r--r--  1 harrietalexander  staff     250 Jul 13 13:05 assembly_score_optimisation.csv\n",
      "-rw-r--r--  1 harrietalexander  staff    1652 Jul 13 13:05 bad.transcripts.fa\n",
      "-rw-r--r--  1 harrietalexander  staff    2163 Jul 13 13:05 contigs.csv\n",
      "-rw-r--r--  1 harrietalexander  staff   27093 Jul 13 13:05 good.transcripts.fa\n",
      "-rw-r--r--  1 harrietalexander  staff       6 Jul 13 13:05 left.fq-right.fq-read_count.txt\n",
      "-rw-r--r--  1 harrietalexander  staff  650074 Jul 13 13:05 left.fq.right.fq.transcripts.assigned.bam\n",
      "-rwxr-----  1 harrietalexander  staff  766295 Jul 13 13:05 left.fq.right.fq.transcripts.bam\n",
      "drwxr-xr-x  3 harrietalexander  staff     102 Jul 13 13:05 libParams\n",
      "drwxr-xr-x  4 harrietalexander  staff     136 Jul 13 13:05 logs\n",
      "drwxr-xr-x  6 harrietalexander  staff     204 Jul 13 13:05 transcripts\n",
      "-rw-r--r--  1 harrietalexander  staff     818 Jul 13 13:05 transcripts.fa_bam_info.csv\n",
      "-rw-r--r--  1 harrietalexander  staff     967 Jul 13 13:05 transcripts.fa_quant.sf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l transrate_results/transcripts/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>contig_name</th>\n",
       "      <th>length</th>\n",
       "      <th>prop_gc</th>\n",
       "      <th>gc_skew</th>\n",
       "      <th>at_skew</th>\n",
       "      <th>cpg_count</th>\n",
       "      <th>cpg_ratio</th>\n",
       "      <th>orf_length</th>\n",
       "      <th>linguistic_complexity_6</th>\n",
       "      <th>in_bridges</th>\n",
       "      <th>p_good</th>\n",
       "      <th>p_bases_covered</th>\n",
       "      <th>p_seq_true</th>\n",
       "      <th>score</th>\n",
       "      <th>p_not_segmented</th>\n",
       "      <th>eff_length</th>\n",
       "      <th>eff_count</th>\n",
       "      <th>tpm</th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0 </th>\n",
       "      <td> NM_001168316</td>\n",
       "      <td> 2283</td>\n",
       "      <td> 0.487516</td>\n",
       "      <td> 0.044025</td>\n",
       "      <td> 0.097436</td>\n",
       "      <td> 177</td>\n",
       "      <td> 1.307349</td>\n",
       "      <td> 533</td>\n",
       "      <td> 0.379150</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.968463</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.889815</td>\n",
       "      <td> 0.918791</td>\n",
       "      <td> 2283</td>\n",
       "      <td>  133.2010</td>\n",
       "      <td>  10886.60</td>\n",
       "      <td>   2.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1 </th>\n",
       "      <td>    NM_174914</td>\n",
       "      <td> 2385</td>\n",
       "      <td> 0.485535</td>\n",
       "      <td> 0.044905</td>\n",
       "      <td> 0.093725</td>\n",
       "      <td> 182</td>\n",
       "      <td> 1.297416</td>\n",
       "      <td> 567</td>\n",
       "      <td> 0.390625</td>\n",
       "      <td> 0</td>\n",
       "      <td> 0.998062</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.999889</td>\n",
       "      <td> 0.954779</td>\n",
       "      <td> 0.956739</td>\n",
       "      <td> 2385</td>\n",
       "      <td> 1577.1500</td>\n",
       "      <td> 123389.00</td>\n",
       "      <td>  33.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2 </th>\n",
       "      <td>    NR_031764</td>\n",
       "      <td> 1853</td>\n",
       "      <td> 0.507285</td>\n",
       "      <td> 0.053191</td>\n",
       "      <td> 0.082147</td>\n",
       "      <td> 154</td>\n",
       "      <td> 1.295481</td>\n",
       "      <td> 221</td>\n",
       "      <td> 0.323975</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.981112</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.941532</td>\n",
       "      <td> 0.959658</td>\n",
       "      <td> 1853</td>\n",
       "      <td>   53.6536</td>\n",
       "      <td>   5402.76</td>\n",
       "      <td>   1.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3 </th>\n",
       "      <td>    NM_004503</td>\n",
       "      <td> 1681</td>\n",
       "      <td> 0.527662</td>\n",
       "      <td>-0.127396</td>\n",
       "      <td> 0.068010</td>\n",
       "      <td> 180</td>\n",
       "      <td> 1.563721</td>\n",
       "      <td> 235</td>\n",
       "      <td> 0.299072</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.998215</td>\n",
       "      <td> 0.999913</td>\n",
       "      <td> 0.900239</td>\n",
       "      <td> 0.901927</td>\n",
       "      <td> 1681</td>\n",
       "      <td>  332.1570</td>\n",
       "      <td>  36869.60</td>\n",
       "      <td>   9.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4 </th>\n",
       "      <td>    NM_006897</td>\n",
       "      <td> 1541</td>\n",
       "      <td> 0.552239</td>\n",
       "      <td>-0.043478</td>\n",
       "      <td> 0.115942</td>\n",
       "      <td> 222</td>\n",
       "      <td> 1.893120</td>\n",
       "      <td> 260</td>\n",
       "      <td> 0.287109</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.999351</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.930615</td>\n",
       "      <td> 0.931219</td>\n",
       "      <td> 1541</td>\n",
       "      <td>  664.0010</td>\n",
       "      <td>  80400.50</td>\n",
       "      <td>  21.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5 </th>\n",
       "      <td>    NM_014212</td>\n",
       "      <td> 2037</td>\n",
       "      <td> 0.566519</td>\n",
       "      <td>-0.067591</td>\n",
       "      <td>-0.023783</td>\n",
       "      <td> 272</td>\n",
       "      <td> 1.671849</td>\n",
       "      <td> 304</td>\n",
       "      <td> 0.347168</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.860579</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.851496</td>\n",
       "      <td> 0.989445</td>\n",
       "      <td> 2037</td>\n",
       "      <td>   55.0092</td>\n",
       "      <td>   5038.92</td>\n",
       "      <td>   1.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6 </th>\n",
       "      <td>    NM_014620</td>\n",
       "      <td> 2300</td>\n",
       "      <td> 0.508261</td>\n",
       "      <td>-0.055603</td>\n",
       "      <td> 0.036251</td>\n",
       "      <td> 225</td>\n",
       "      <td> 1.519449</td>\n",
       "      <td> 264</td>\n",
       "      <td> 0.366455</td>\n",
       "      <td> 0</td>\n",
       "      <td> 0.998382</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.999815</td>\n",
       "      <td> 0.911184</td>\n",
       "      <td> 0.912830</td>\n",
       "      <td> 2300</td>\n",
       "      <td>  609.5190</td>\n",
       "      <td>  49448.40</td>\n",
       "      <td>  13.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7 </th>\n",
       "      <td>    NM_017409</td>\n",
       "      <td> 1959</td>\n",
       "      <td> 0.533435</td>\n",
       "      <td>-0.100478</td>\n",
       "      <td> 0.094092</td>\n",
       "      <td> 227</td>\n",
       "      <td> 1.645488</td>\n",
       "      <td> 342</td>\n",
       "      <td> 0.340576</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.873915</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.867795</td>\n",
       "      <td> 0.992997</td>\n",
       "      <td> 1959</td>\n",
       "      <td>   47.0093</td>\n",
       "      <td>   4477.57</td>\n",
       "      <td>   1.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8 </th>\n",
       "      <td>    NM_017410</td>\n",
       "      <td> 2396</td>\n",
       "      <td> 0.575960</td>\n",
       "      <td>-0.059420</td>\n",
       "      <td> 0.037402</td>\n",
       "      <td> 339</td>\n",
       "      <td> 1.712082</td>\n",
       "      <td> 330</td>\n",
       "      <td> 0.388916</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.769199</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.765217</td>\n",
       "      <td> 0.994823</td>\n",
       "      <td> 2396</td>\n",
       "      <td>   42.0094</td>\n",
       "      <td>   3271.54</td>\n",
       "      <td>   0.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9 </th>\n",
       "      <td>    NM_018953</td>\n",
       "      <td> 1612</td>\n",
       "      <td> 0.591811</td>\n",
       "      <td>-0.121593</td>\n",
       "      <td> 0.051672</td>\n",
       "      <td> 244</td>\n",
       "      <td> 1.754636</td>\n",
       "      <td> 229</td>\n",
       "      <td> 0.297852</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.998139</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.925380</td>\n",
       "      <td> 0.927105</td>\n",
       "      <td> 1612</td>\n",
       "      <td>  228.0070</td>\n",
       "      <td>  26392.20</td>\n",
       "      <td>   7.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>    NM_022658</td>\n",
       "      <td> 2288</td>\n",
       "      <td> 0.482517</td>\n",
       "      <td>-0.050725</td>\n",
       "      <td> 0.087838</td>\n",
       "      <td> 189</td>\n",
       "      <td> 1.422848</td>\n",
       "      <td> 242</td>\n",
       "      <td> 0.370850</td>\n",
       "      <td> 0</td>\n",
       "      <td> 0.999590</td>\n",
       "      <td> 0.999126</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.932470</td>\n",
       "      <td> 0.933668</td>\n",
       "      <td> 2288</td>\n",
       "      <td> 4880.9400</td>\n",
       "      <td> 398052.00</td>\n",
       "      <td> 106.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>    NM_153633</td>\n",
       "      <td> 1666</td>\n",
       "      <td> 0.493998</td>\n",
       "      <td>-0.115431</td>\n",
       "      <td> 0.100830</td>\n",
       "      <td> 161</td>\n",
       "      <td> 1.605413</td>\n",
       "      <td> 279</td>\n",
       "      <td> 0.294922</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.999613</td>\n",
       "      <td> 0.914682</td>\n",
       "      <td> 0.915036</td>\n",
       "      <td> 1666</td>\n",
       "      <td>  343.6030</td>\n",
       "      <td>  38483.60</td>\n",
       "      <td>  10.31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>    NM_153693</td>\n",
       "      <td> 2072</td>\n",
       "      <td> 0.527992</td>\n",
       "      <td>-0.060329</td>\n",
       "      <td> 0.014315</td>\n",
       "      <td> 219</td>\n",
       "      <td> 1.522098</td>\n",
       "      <td> 153</td>\n",
       "      <td> 0.339355</td>\n",
       "      <td> 0</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.978282</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.942954</td>\n",
       "      <td> 0.963888</td>\n",
       "      <td> 2072</td>\n",
       "      <td>   71.7407</td>\n",
       "      <td>   6460.54</td>\n",
       "      <td>   1.73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>    NM_173860</td>\n",
       "      <td>  849</td>\n",
       "      <td> 0.666667</td>\n",
       "      <td>-0.007067</td>\n",
       "      <td> 0.024735</td>\n",
       "      <td> 203</td>\n",
       "      <td> 2.152051</td>\n",
       "      <td> 282</td>\n",
       "      <td> 0.164795</td>\n",
       "      <td> 0</td>\n",
       "      <td> 0.998960</td>\n",
       "      <td> 0.998822</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.814451</td>\n",
       "      <td> 0.816260</td>\n",
       "      <td>  849</td>\n",
       "      <td>  961.9970</td>\n",
       "      <td> 211426.00</td>\n",
       "      <td>  56.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>    NR_003084</td>\n",
       "      <td> 1640</td>\n",
       "      <td> 0.560366</td>\n",
       "      <td>-0.044614</td>\n",
       "      <td>-0.051318</td>\n",
       "      <td> 206</td>\n",
       "      <td> 1.603265</td>\n",
       "      <td> 182</td>\n",
       "      <td> 0.289551</td>\n",
       "      <td> 0</td>\n",
       "      <td> 0.000000</td>\n",
       "      <td> 0.060976</td>\n",
       "      <td> 1.000000</td>\n",
       "      <td> 0.010000</td>\n",
       "      <td> 0.997424</td>\n",
       "      <td> 1640</td>\n",
       "      <td>    0.0000</td>\n",
       "      <td>      0.00</td>\n",
       "      <td>   0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     contig_name  length   prop_gc   gc_skew   at_skew  cpg_count  cpg_ratio  \\\n",
       "0   NM_001168316    2283  0.487516  0.044025  0.097436        177   1.307349   \n",
       "1      NM_174914    2385  0.485535  0.044905  0.093725        182   1.297416   \n",
       "2      NR_031764    1853  0.507285  0.053191  0.082147        154   1.295481   \n",
       "3      NM_004503    1681  0.527662 -0.127396  0.068010        180   1.563721   \n",
       "4      NM_006897    1541  0.552239 -0.043478  0.115942        222   1.893120   \n",
       "5      NM_014212    2037  0.566519 -0.067591 -0.023783        272   1.671849   \n",
       "6      NM_014620    2300  0.508261 -0.055603  0.036251        225   1.519449   \n",
       "7      NM_017409    1959  0.533435 -0.100478  0.094092        227   1.645488   \n",
       "8      NM_017410    2396  0.575960 -0.059420  0.037402        339   1.712082   \n",
       "9      NM_018953    1612  0.591811 -0.121593  0.051672        244   1.754636   \n",
       "10     NM_022658    2288  0.482517 -0.050725  0.087838        189   1.422848   \n",
       "11     NM_153633    1666  0.493998 -0.115431  0.100830        161   1.605413   \n",
       "12     NM_153693    2072  0.527992 -0.060329  0.014315        219   1.522098   \n",
       "13     NM_173860     849  0.666667 -0.007067  0.024735        203   2.152051   \n",
       "14     NR_003084    1640  0.560366 -0.044614 -0.051318        206   1.603265   \n",
       "\n",
       "    orf_length  linguistic_complexity_6  in_bridges    p_good  \\\n",
       "0          533                 0.379150           0  1.000000   \n",
       "1          567                 0.390625           0  0.998062   \n",
       "2          221                 0.323975           0  1.000000   \n",
       "3          235                 0.299072           0  1.000000   \n",
       "4          260                 0.287109           0  1.000000   \n",
       "5          304                 0.347168           0  1.000000   \n",
       "6          264                 0.366455           0  0.998382   \n",
       "7          342                 0.340576           0  1.000000   \n",
       "8          330                 0.388916           0  1.000000   \n",
       "9          229                 0.297852           0  1.000000   \n",
       "10         242                 0.370850           0  0.999590   \n",
       "11         279                 0.294922           0  1.000000   \n",
       "12         153                 0.339355           0  1.000000   \n",
       "13         282                 0.164795           0  0.998960   \n",
       "14         182                 0.289551           0  0.000000   \n",
       "\n",
       "    p_bases_covered  p_seq_true     score  p_not_segmented  eff_length  \\\n",
       "0          0.968463    1.000000  0.889815         0.918791        2283   \n",
       "1          1.000000    0.999889  0.954779         0.956739        2385   \n",
       "2          0.981112    1.000000  0.941532         0.959658        1853   \n",
       "3          0.998215    0.999913  0.900239         0.901927        1681   \n",
       "4          0.999351    1.000000  0.930615         0.931219        1541   \n",
       "5          0.860579    1.000000  0.851496         0.989445        2037   \n",
       "6          1.000000    0.999815  0.911184         0.912830        2300   \n",
       "7          0.873915    1.000000  0.867795         0.992997        1959   \n",
       "8          0.769199    1.000000  0.765217         0.994823        2396   \n",
       "9          0.998139    1.000000  0.925380         0.927105        1612   \n",
       "10         0.999126    1.000000  0.932470         0.933668        2288   \n",
       "11         1.000000    0.999613  0.914682         0.915036        1666   \n",
       "12         0.978282    1.000000  0.942954         0.963888        2072   \n",
       "13         0.998822    1.000000  0.814451         0.816260         849   \n",
       "14         0.060976    1.000000  0.010000         0.997424        1640   \n",
       "\n",
       "    eff_count        tpm  coverage  \n",
       "0    133.2010   10886.60      2.92  \n",
       "1   1577.1500  123389.00     33.06  \n",
       "2     53.6536    5402.76      1.45  \n",
       "3    332.1570   36869.60      9.88  \n",
       "4    664.0010   80400.50     21.54  \n",
       "5     55.0092    5038.92      1.35  \n",
       "6    609.5190   49448.40     13.25  \n",
       "7     47.0093    4477.57      1.20  \n",
       "8     42.0094    3271.54      0.88  \n",
       "9    228.0070   26392.20      7.07  \n",
       "10  4880.9400  398052.00    106.66  \n",
       "11   343.6030   38483.60     10.31  \n",
       "12    71.7407    6460.54      1.73  \n",
       "13   961.9970  211426.00     56.65  \n",
       "14     0.0000       0.00      0.00  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "contigs=pd.read_csv(\"transrate_results/transcripts/contigs.csv\")\n",
    "contigs\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Looking at pre-existing data from NCBI\n",
    "\n",
    "<img src=\"images/Spearman.png\">\n",
    "\n",
    "<img src=\"images/Comparison4.png\">\n",
    "\n",
    "<img src=\"images/AllData.png\">\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
